tools/emscripten/pytest_scikit_image_pyodide.js

```diff
@@ -0,0 +1,57 @@
+// A JavaScript file to run the scikit-image test suite using Pyodide
+// This file is used by the GitHub Actions workflow to run the tests
+// against the Pyodide build of scikit-image defined in emscripten.yml.
+
+// The contents of this file are attributed to the scikit-learn developers,
+// who have a similar file in their repository:
+// https://github.com/scikit-learn/scikit-learn/blob/main/build_tools/azure/pytest-pyodide.js
+
+
+const { opendir } = require('node:fs/promises');
+const { loadPyodide } = require("pyodide");
+
+async function main() {
+    let exit_code = 0;
+    try {
+        global.pyodide = await loadPyodide();
+        let pyodide = global.pyodide;
+        const FS = pyodide.FS;
+        const NODEFS = FS.filesystems.NODEFS;
+
+        let mountDir = "/mnt";
+        pyodide.FS.mkdir(mountDir);
+        pyodide.FS.mount(pyodide.FS.filesystems.NODEFS, { root: "." }, mountDir);
+
+        await pyodide.loadPackage(["micropip"]);
+        await pyodide.runPythonAsync(`
+            import glob
+            import micropip
+
+            wheels = glob.glob("/mnt/dist/*.whl")
+            wheels = [f'emfs://{wheel}' for wheel in wheels]
+            print(f"Installing wheels: {wheels}")
+            await micropip.install(wheels);
+
+            pkg_list = micropip.list()
+            print(pkg_list)
+        `);
+
+
+        await pyodide.runPythonAsync("import micropip; micropip.install('pytest')");
+        await pyodide.runPythonAsync("import micropip; micropip.install('matplotlib')");
```

Comment:
I don't particularly like that installing these extra test dependencies is tucked away here. 

Do you think it would be feasible to try installing the dependencies in `requirements/test.yml` and just ignore deps that don't work? Alternatively, we could add an extra `requirements/pyodide_test.txt`.

Reply:
You're right, we could do better with less repetition – I can indeed add a `requirements.txt` file, but we still don't have the necessary functionality to install from such files in `micropip` yet: https://github.com/pyodide/micropip/issues/6, i.e., the file will remain redundant for a while.

I couldn't find a`requirements/test.yml` file as you suggested, so I assume you meant the `requirements/test.txt` file – out of these:

```txt
# Generated via tools/generate_requirements.py and pre-commit hook.
# Do not edit this file; modify pyproject.toml instead.
asv
numpydoc>=1.7
pooch>=1.6.0
pytest>=7.0
pytest-cov>=2.11.0
pytest-localserver
pytest-faulthandler
pytest-doctestplus
```

dependencies, `pytest` and `pytest-cov` are likely the only ones that would work right now, and `asv` and `pooch` serve different uses that are not of use here I guess. The rest of the plugins make more sense to be running in a non-Pyodide environment and outside of the regular tests.

Reply:
Okay, then let's keep streamlining this for a later day. We can't run the full test suite anyway yet.